### DISCUSSION TITLE
Kernel Panic

### DISCUSSION THREAD

**Discussion by partisani** [CosmosOS/Cosmos#1956] [Ideas]
How Do I Make A Handler For When The Operating System Crash It Shows A Kernel _"Panik"_ Screen

And A Kernel Panic Debugger For Manually Crash It!

**Comment by zarlo** [comment]
all of this can be done, it just not built into cosmos
if you want to see how it cn be done https://github.com/aura-systems/Aura-Operating-System/blob/master/Aura%20Operating%20System/Aura_Plugs/Core/INTs.cs#L16 is a good starting point

**Reply by partisani** [reply]
I'm A Beginner So I Don't Have Understand

**Comment by prescott2m** [comment]
i always did `try { }` and `catch(Exception e) { }` in the kernel file but there is no built in way to do this.

**Reply by quajak** [reply]
You cant currently recover from a kernel panic, since kernel panics mean that cosmos doesnt understand what went wrong, only that something went terribly wrong such as stack corruption...

The goal is to ensure that this behaviour never happens, so i dont know how much effort should be put into handling when it happens

**Comment by prescott2m** [comment]
i just want to know the kernel panic codes.

**Reply by quajak** [reply]
Which ones?

**Reply by prescott2m** [reply]
0x2

**Reply by quajak** [reply]
I dont recall every seeing that one.
